In [3]:
import pandas as pd

lab = pd.read_csv("lab_results_clean.csv")
billing = pd.read_csv("billing_clean.csv")

print(lab.head())
print(billing.head())

   Lab_ID Admission_ID         Test   Value
0  L00001       A00001  Blood Sugar   91.27
1  L00002       A00002  Blood Sugar  175.69
2  L00003       A00003  Cholesterol  150.60
3  L00004       A00004  Cholesterol  176.53
4  L00005       A00005  Blood Sugar  133.46
  Bill_ID Admission_ID Patient_ID  Total  Insurance   Paid  Pending
0  B00001       A00001     P00001  70078         93  70001       77
1  B00002       A00002     P00002  20824       4403  17971     2853
2  B00003       A00003     P00003   6432        239   6228      204
3  B00004       A00004     P00004  82670       9986  77817     4853
4  B00005       A00005     P00005  10376       4009   7571     2805


In [4]:
# Treatment / Service Demand

test_demand = lab["Test"].value_counts()

print(test_demand)

Test
Blood Sugar    2567
Cholesterol    2433
Name: count, dtype: int64


In [5]:
# Merge lab results with billing data using Admission_ID

merged = pd.merge(
    lab,
    billing,
    on="Admission_ID",
    how="inner"
)

print(merged.head())
print("Number of records:", len(merged))

   Lab_ID Admission_ID         Test   Value Bill_ID Patient_ID  Total  \
0  L00001       A00001  Blood Sugar   91.27  B00001     P00001  70078   
1  L00002       A00002  Blood Sugar  175.69  B00002     P00002  20824   
2  L00003       A00003  Cholesterol  150.60  B00003     P00003   6432   
3  L00004       A00004  Cholesterol  176.53  B00004     P00004  82670   
4  L00005       A00005  Blood Sugar  133.46  B00005     P00005  10376   

   Insurance   Paid  Pending  
0         93  70001       77  
1       4403  17971     2853  
2        239   6228      204  
3       9986  77817     4853  
4       4009   7571     2805  
Number of records: 5000


In [6]:
# Billing by laboratory test

billing_by_test = (
    merged.groupby("Test")["Total"]
    .agg(["count", "sum", "mean"])
    .sort_values("sum", ascending=False)
)

print(billing_by_test)

             count        sum          mean
Test                                       
Blood Sugar   2567  135814788  52907.981301
Cholesterol   2433  127578317  52436.628442


In [8]:
# Overall Billing KPIs

total_billing = billing["Total"].sum()
total_insurance = billing["Insurance"].sum()
total_paid = billing["Paid"].sum()
total_pending = billing["Pending"].sum()

print("Total Billing:", f"{total_billing:,}")
print("Total Insurance:", f"{total_insurance:,}")
print("Total Paid:", f"{total_paid:,}")
print("Total Pending:", f"{total_pending:,}")

Total Billing: 263,393,105
Total Insurance: 66,703,846
Total Paid: 229,458,756
Total Pending: 33,934,349


In [9]:
# Payment and Pending Rates

paid_percentage = (total_paid / total_billing) * 100
pending_percentage = (total_pending / total_billing) * 100

print("Paid Percentage:", round(paid_percentage, 2), "%")
print("Pending Percentage:", round(pending_percentage, 2), "%")

Paid Percentage: 87.12 %
Pending Percentage: 12.88 %


In [ ]:
# Member 7 - Treatment & Service Demand KPI Summary

# Most demanded test
most_demanded_test = lab["Test"].value_counts().idxmax()
most_demanded_count = lab["Test"].value_counts().max()

# Test demand
test_demand = lab["Test"].value_counts()

# Billing by test
billing_by_test = (
    merged.groupby("Test")["Total"]
    .agg(["count", "sum", "mean"])
    .sort_values("sum", ascending=False)
)

# Overall billing
total_billing = billing["Total"].sum()
total_paid = billing["Paid"].sum()
total_pending = billing["Pending"].sum()

# Payment percentages
paid_percentage = (total_paid / total_billing) * 100
pending_percentage = (total_pending / total_billing) * 100


# Display results
print("===== TREATMENT & SERVICE DEMAND =====")
print()

print("Most Demanded Test:", most_demanded_test)
print("Number of Tests:", f"{most_demanded_count:,}")
print()

print("Blood Sugar Tests:", f"{test_demand['Blood Sugar']:,}")
print("Cholesterol Tests:", f"{test_demand['Cholesterol']:,}")
print()

print("Blood Sugar Billing: ₹", f"{billing_by_test.loc['Blood Sugar', 'sum']:,}")
print("Cholesterol Billing: ₹", f"{billing_by_test.loc['Cholesterol', 'sum']:,}")
print()

print("Total Billing: ₹", f"{total_billing:,}")
print("Total Paid: ₹", f"{total_paid:,}")
print("Total Pending: ₹", f"{total_pending:,}")
print()

print("Paid Percentage:", round(paid_percentage, 2), "%")
print("Pending Percentage:", round(pending_percentage, 2), "%")




===== TREATMENT & SERVICE DEMAND =====

Most Demanded Test: Blood Sugar
Number of Tests: 2,567

Blood Sugar Tests: 2,567
Cholesterol Tests: 2,433

Blood Sugar Billing: ₹ 135,814,788
Cholesterol Billing: ₹ 127,578,317

Total Billing: ₹ 263,393,105
Total Paid: ₹ 229,458,756
Total Pending: ₹ 33,934,349

Paid Percentage: 87.12 %
Pending Percentage: 12.88 %


# Member 7 - Treatment & Service Demand Analysis

## Key Findings

- Blood Sugar was the most demanded laboratory test with 2,567 tests.
- Cholesterol accounted for 2,433 tests.
- Blood Sugar generated the highest total billing of ₹135,814,788.
- Cholesterol generated ₹127,578,317 in total billing.
- Overall billing amounted to ₹263,393,105.
- ₹229,458,756 was paid.
- ₹33,934,349 remained pending.
- 87.12% of total billing was paid.
- 12.88% of total billing remained pending.

## Conclusion

Blood Sugar was the most demanded laboratory test and also generated the highest
total billing. The hospital recorded ₹263,393,105 in total billing, of which
87.12% was paid and 12.88% remained pending.